operator/internal/manifests/distributor.go

```diff
@@ -117,10 +123,11 @@ func NewDistributorDeployment(opts Options) *appsv1.Deployment {
 				TerminationMessagePath:   "/dev/termination-log",
 				TerminationMessagePolicy: "File",
 				ImagePullPolicy:          "IfNotPresent",
-				SecurityContext:          containerSecurityContext(),
+				SecurityContext: &corev1.SecurityContext{
+					AllowPrivilegeEscalation: pointer.Bool(false),
+				},
```

Comment:
Why not moving such repetition into `configurePodSpecForRestrictedStandard` ?

Reply:
So, this is just ensuring that the containers will not allow privilege escalation at all times since it's an opt in to make restrictive. Not sure if we only want to toggle this on when building for `restricted` or not.

Reply:
@periklis , I have changed it so the `allowPrivilegeEscalation: false` is only set when the restricted feature flag is on. I have also applied this logic to only apply to the OpenShift builds through the overlay.